weldx/tests/asdf_tests/test_asdf_util.py

```diff
@@ -163,14 +163,19 @@ class _DataClass:
 
 def test_write_buffer_dummy_inline_arrays():
     """Test dummy inline arrays argument for write_buffer."""
+    from asdf.tagged import TaggedDict
+
     name = "large_array"
     array = np.random.random(50)
     buff = write_buffer(tree={name: array}, write_kwargs=dict(dummy_arrays=True))
 
-    buff.seek(0)
-    restored = read_buffer(buff)[name]
-    assert restored.dtype == array.dtype
-    assert restored.shape == array.shape
+    with read_buffer_context(buff) as data:
+        restored = data[name]
+        assert type(restored) == TaggedDict
+        assert restored._tag.startswith("tag:stsci.edu:asdf/core/ndarray-")
+        assert restored["data"] == []
+        assert restored["datatype"] == array.dtype
+        assert restored["shape"][0] == array.shape[0]
```

Comment:
looking at the old implementation I am surprised that this seemingly converted to `ndarray` previously @braingram 

The current behavior with a warning and reading as raw `TaggedDict` seems more fitting so I adjusted this test for now to reflect that expected output

Reply:
Yeah, it's unclear what should be returned as asdf appears to have issues with empty inline arrays:
https://github.com/asdf-format/asdf/issues/1538
Do you have a link to the run that returned the `TaggedDict`? I don't think I saw that while running locally.

Reply:
It looks like, if we run the tests without the `--weldx-file-rw-buffer --weldx-file-rw-buffer-disp-header` flag, a  `TaggedDict` is returned (cannot convert to `ndarray`)
See an example here: https://github.com/BAMWelDX/weldx/actions/runs/4935769622/jobs/8822498546

However, if `--weldx-file-rw-buffer --weldx-file-rw-buffer-disp-header` is set, it looks like we do get an `ndarray`
example: https://github.com/BAMWelDX/weldx/actions/runs/4947440349/jobs/8846812586?pr=875

Do you have any idea here @marscher ?

Reply:
oddly I cannot reproduce this locally. I will have to dig into it a bit.

Reply:
I just removed this code branch. It just leads to confusion and is not really used.